scu_builders.py

```diff
@@ -0,0 +1,335 @@
+"""Functions used to generate scu build source files during build time
```

Comment:
I'm not sure what is the Godot stance on typing for Python (@akien-mga are we against it to minimize Python knowledge requirements ?)

But as a general guideline, I'd would advice you adding type to the function parameters and return. This should be fairly simple as we don't use complex types (e.g. `def my_func(base_dir: Path, module_name: str) -> None`) and improve the readability a lot (on top of helping your IDE detect errors) ;-)

Reply:
We started using typing in a few files in `master`, it's ok. Our minimal required Python version is currently 3.6 which is the one adding type hints.

In the `3.x` branch we still support Python 3.5 in theory so we typically don't use type hints there (though I see we've started using them for `doc/tools/make_rst.py`, which is fine as it's only used by documentation maintainers, and in `methods.py::generate_vs_project`, which might be more debatable but nobody complained so far...